### PR TITLE
[Notifier] Remove deprecation in slack-notifier

### PR DESCRIPTION
**Pull request by jschaedl**
| Q             | A
| ------------- | ---
| Branch?       | 6.0
| Bug fix?      | no
| New feature?  | no 
| Deprecations? | no 
| Tickets       | - 
| License       | MIT
| Doc PR        | -

### CODE REVIEW

File: UPGRADE-6.0.md [hunk 1]
```diff
@@ -151,6 +151,11 @@ Monolog
  * The `$actionLevel` constructor argument of `Symfony\Bridge\Monolog\Handler\FingersCrossed\NotFoundActivationStrategy` has been replaced by the `$inner` one which expects an ActivationStrategyInterface to decorate instead. `Symfony\Bridge\Monolog\Handler\FingersCrossed\NotFoundActivationStrategy` is now final.
  * The `$actionLevel` constructor argument of `Symfony\Bridge\Monolog\Handler\FingersCrossed\HttpCodeActivationStrategy` has been replaced by the `$inner` one which expects an ActivationStrategyInterface to decorate instead. `Symfony\Bridge\Monolog\Handler\FingersCrossed\HttpCodeActivationStrategy` is now final.
 
+Notifier
+--------
+
+ * symfony/slack-notifier: Removed `SlackOptions::channel()`, use `SlackOptions::recipient()` instead.
```
Comment (derrabus) [inline UPGRADE-6.0.md]: This message should have been added with the PR that *introduced* the deprecation. 🤔 

Can you please submit this line as a PR to the 5.2 branch?

File: src/Symfony/Component/Notifier/Bridge/Slack/Tests/SlackOptionsTest.php [hunk 2]
```diff
@@ -122,16 +122,6 @@ public function testSetBlock()
         $this->assertSame([['type' => 'divider']], $options->toArray()['blocks']);
     }
 
-    /**
-     * @group legacy
-     */
-    public function testChannelMethodRaisesDeprecation()
-    {
-        $this->expectDeprecation('Since symfony/slack-notifier 5.1: The "Symfony\Component\Notifier\Bridge\Slack\SlackOptions::channel()" method is deprecated, use "recipient()" instead.');
```
Comment (derrabus) [inline src/Symfony/Component/Notifier/Bridge/Slack/Tests/SlackOptionsTest.php:130]: Please also remove `ExpectDeprecationTrait` from this class.

File: src/Symfony/Component/Notifier/Bridge/Slack/SlackOptions.php [hunk 3]
```diff
@@ -58,18 +58,6 @@ public function getRecipientId(): ?string
...
     }
 
-    /**
-     * @return $this
-     *
-     * @deprecated since Symfony 5.1, use recipient() instead.
-     */
-    public function channel(string $channel): self
-    {
-        trigger_deprecation('symfony/slack-notifier', '5.1', 'The "%s()" method is deprecated, use "recipient()" instead.', __METHOD__);
```
Comment (derrabus) [inline src/Symfony/Component/Notifier/Bridge/Slack/SlackOptions.php:68]: Please remove `symfony/deprecation-contracts` from the composer.json file of the bridge.

**Review by derrabus — CHANGES_REQUESTED** [review]

**Review by jderusse — COMMENTED** [review]
I wonder if we should also remove this code 

https://github.com/symfony/symfony/blob/a2c8e84b7e8618c814458d7c64e56d8b9b779639/src/Symfony/Component/Notifier/Bridge/Slack/SlackTransportFactory.php#L34-L36

**Comment by jschaedl** [comment]
> I wonder if we should also remove this code
> 
> https://github.com/symfony/symfony/blob/a2c8e84b7e8618c814458d7c64e56d8b9b779639/src/Symfony/Component/Notifier/Bridge/Slack/SlackTransportFactory.php#L34-L36

I think that make sense. This is missing an entry in the changelog of 5.2 too. Shall I add that entry in a separate PR against 5.2?

**Comment by jschaedl** [comment]
> > I wonder if we should also remove this code
> > https://github.com/symfony/symfony/blob/a2c8e84b7e8618c814458d7c64e56d8b9b779639/src/Symfony/Component/Notifier/Bridge/Slack/SlackTransportFactory.php#L34-L36
> 
> I think that make sense. This is missing an entry in the changelog of 5.2 too. Shall I add that entry in a separate PR against 5.2?

I checked it again. It's probably this entry: https://github.com/symfony/symfony/blame/4bf9461d9b0d2ee6840f4e3873542f8b54a5f8f4/src/Symfony/Component/Notifier/Bridge/Slack/CHANGELOG.md#L21

File: UPGRADE-6.0.md (hunk 1 above)
Comment (jschaedl) [reply UPGRADE-6.0.md]: Here you go: #41309

File: src/Symfony/Component/Notifier/Bridge/Slack/CHANGELOG.md [hunk 4]
```diff
@@ -1,6 +1,11 @@
 CHANGELOG
 =========
 
+6.0
+---
+
+ * Removed `SlackOptions::channel()`, use `SlackOptions::recipient()` instead.
```
Comment (OskarStark) [inline src/Symfony/Component/Notifier/Bridge/Slack/CHANGELOG.md]:
```suggestion
 * Remove `SlackOptions::channel()`, use `SlackOptions::recipient()` instead.
```

**Review by OskarStark — APPROVED** [review]

File: src/Symfony/Component/Notifier/Bridge/Slack/CHANGELOG.md [hunk 5]
```diff
@@ -1,6 +1,11 @@
 CHANGELOG
 =========
 
+6.0
+---
+
+ * Remove `SlackOptions::channel()`, use `SlackOptions::recipient()` instead.
```
Comment (nicolas-grekas) [inline src/Symfony/Component/Notifier/Bridge/Slack/CHANGELOG.md]: no final dot please

**Comment by nicolas-grekas** [comment]
Thank you @jschaedl.

**Pull request merged**